chain/chain/src/store.rs

```diff
@@ -546,10 +546,34 @@ impl ChainStoreAccess for ChainStore {
 
     /// Get full block.
     fn get_block(&mut self, h: &CryptoHash) -> Result<&Block, Error> {
-        option_to_not_found(
+        let block_result = option_to_not_found(
             read_with_cache(&*self.store, ColBlock, &mut self.blocks, h.as_ref()),
             &format!("BLOCK: {}", h),
-        )
+        );
+        match block_result {
+            Ok(block) => Ok(block),
+            Err(e) => match e.kind() {
+                ErrorKind::DBNotFoundErr(_) => {
+                    let block_header_result = read_with_cache(
+                        &*self.store,
+                        ColBlockHeader,
+                        &mut self.headers,
+                        h.as_ref(),
+                    );
+                    match block_header_result {
+                        Ok(Some(_)) => Err(ErrorKind::BlockMissing(h.clone()).into()),
+                        Ok(None) => Err(e),
+                        Err(header_error) => {
```

Comment:
No need to do this. It might cause node to crash unnecessarily.

Reply:
Good point! Downgraded to a debug message + debug assert.